src/dokan/ceph_dokan.c

```diff
@@ -0,0 +1,1489 @@
+/*
+  A CephFS Client on Win32 (based on Dokan)
+*/
+
+#define UNICODE
+#define _UNICODE
+
+#include "include/compat.h"
+#include "include/cephfs/libcephfs.h"
+
+#include <stdio.h>
+#include <stdlib.h>
+
+#include <dokan.h>
+#include <fileinfo.h>
+
+#include "dbg.h"
+#include "posix_acl.h"
+#include "utils.h"
+
+#include <dirent.h>
+#include <fcntl.h>
+#include <signal.h>
+#include <sddl.h>
+
+#include <accctrl.h>
+#include <aclapi.h>
+
+#define MAX_PATH_CEPH 8192
+#define CEPH_DOKAN_IO_DEFAULT_TIMEOUT 1000 * 60 * 5
+
+#define READ_ACCESS_REQUESTED(access_mode) \
+    access_mode & GENERIC_READ || \
+    access_mode & FILE_SHARE_READ || \
+    access_mode & STANDARD_RIGHTS_READ || \
+    access_mode & FILE_SHARE_READ
+#define WRITE_ACCESS_REQUESTED(access_mode) \
+    access_mode & GENERIC_WRITE || \
+    access_mode & FILE_SHARE_WRITE || \
+    access_mode & STANDARD_RIGHTS_WRITE || \
+    access_mode & FILE_SHARE_WRITE
```

Comment:
These #define's need to get the boolean value enclosed in parentheses -- I know they're used safely here, but it'd be easy for somebody to stick them inside a compound statement and have very bad things happen.